### ISSUE TITLE
Warn about implicit string concatenation in a list literal

### ISSUE THREAD

**Issue by noamraph** [astral-sh/ruff#28316]
### Summary

I had this issue a few times, so I think it's a good candidate for a linter rule. I have a list like this:

```python
EXCLUDE_PATTERNS = [
    "bin/**",
    "lib/python*/ensurepip/**",
    "lib/python*/idlelib/**",
    "libexec/**"
    "sbin/**",
    "share/doc/**",
]
```

It's really hard to notice that the list doesn't include the patterns `"libexec/**"` and `"sbin/**"`, but rather the pattern `"libexec/**sbin/**"`, since the strings are implicitly concatenated.

Perhaps the rule should be: warn about implicit string concatenation inside a list literal.

I can try to design and implement this.

WDYT?

**Comment by AlexWaygood** [comment]
Good news! This already exists: https://docs.astral.sh/ruff/rules/implicit-string-concatenation-in-collection-literal/
